dislib/data/array.py

```diff
@@ -325,9 +325,20 @@ def _get_slice(self, rows, cols):
                                       " the dislib team or open an issue "
                                       "in github.")
 
+        if r_start >= r_stop or c_start >= c_stop:
+            shape = [0, 0]
+            if r_start < r_stop:
+                shape[0] = r_stop - r_start
+            if c_start < c_stop:
+                shape[1] = c_stop - c_start
+            res = Array(blocks=[[np.empty(shape)]], top_left_shape=shape,
+                        reg_shape=self._reg_shape, shape=shape,
+                        sparse=self._sparse)
+            return res
+
```

Comment:
The previous implementation returned arrays with empty blocks at the end. I corrected that as it was unnecessary, and caused difficulties when working with array blocks. As a result of the correction, empty slices no longer returned empty blocks, and thus failed the Array._validate_blocks() (raise AttributeError('Blocks must a list of lists, with at least an empty numpy/scipy matrix.')). Although I don't agree with this validation, this is a matter of discussion, so I added this code block to handle this special case in the most natural way possible (having the empty numpy array keep the number of rows/cols when possible). However I didn't consider to keep the sparsity of the array, and this starts to be too annoying.

Anyways, I'm going to add a short comment to explain what does this block do.

Reply:
Ok, I understand. Just a couple of comments:

1. Not sure if it matters, but shape should probably be a tuple instead of a list
2. If the array is empty maybe it is better to set sparse to False in all cases
3. Are `top_left_shape` and `reg_shape` defined in consistency with other slicing cases?

Reply:
I have modified this block, to simplify. Now the empty block will always have a shape of (0,0) (if this causes any problem, we probably have to be more explicit when defining what is a valid Array).
1. Solved.
2. I think it's better to keep it sparse if the original array is sparse, for consistency with other slices.
3. `reg_shape` is the same as in the original array, and it is consistent with other slices. `top_left_shape` is not consistent with other slices, but I think it doesn't need to be, we can reset it to `top_left_shape = reg_shape`, which is consistent with the default behavior of the array constructor when creating a new Array.

Reply:
I don't know what happens in other cases, but if the contents of the array is a numpy array, sparse should be False for consistency. An alternative solution is to create the array with an empty CSR matrix if sparse is True.

Also, now you can do the following to avoid the if statements:

```
nrows = max(0, nrows)
ncols = max(0, ncols)
```

Reply:
I added the creation of an empty CSR matrix if sparse is True, but I hope in the future we remove the empty_block requirement and we don't need to do this.

I also applied your proposed change for `n_rows` and `n_cols.`